### PR TITLE
Display outputs last in diff view

### LINKED ISSUE #3400: Outputs in diff view can be printed out of order
I have a program whose outputs show up out of order in the "details" diff view:

```
Previewing update (dev):

     Type                    Name              Plan       Info
     pulumi:pulumi:Stack     iac-workshop-dev
 ~   ├─ aws:s3:Bucket        my-bucket         update     [diff: +website]
 ~   └─ aws:s3:BucketObject  index.html        update     [diff: ~acl,contentType]

Outputs:
  + bucketEndpoint: output<string>
  ~ bucketName    : "my-bucket-02d7e7a" => output<string>

Resources:
    ~ 2 to update
    1 unchanged

Do you want to perform this update? details
  pulumi:pulumi:Stack: (same)
    [urn=urn:pulumi:dev::iac-workshop::pulumi:pulumi:Stack::iac-workshop-dev]
    ~ aws:s3/bucket:Bucket: (update)
        [id=my-bucket-02d7e7a]
        [urn=urn:pulumi:dev::iac-workshop::aws:s3/bucket:Bucket::my-bucket]
        [provider=urn:pulumi:dev::iac-workshop::pulumi:providers:aws::default_1_7_0::229428dd-3bcc-4dcf-ae56-ded0b3b0f322]
      + website: {
          + indexDocument: "index.html"
        }
    --outputs:--
  + bucketEndpoint: output<string>
  ~ bucketName    : "my-bucket-02d7e7a" => output<string>
    ~ aws:s3/bucketObject:BucketObject: (update)
        [id=index.html]
        [urn=urn:pulumi:dev::iac-workshop::aws:s3/bucketObject:BucketObject::index.html]
        [provider=urn:pulumi:dev::iac-workshop::pulumi:providers:aws::default_1_7_0::229428dd-3bcc-4dcf-ae56-ded0b3b0f322]
      ~ acl        : "private" => "public-read"
      ~ contentType: "binary/octet-stream" => "text/html"

Do you want to perform this update? no
```

Notice the `--outputs:--` section is intermixed amongst the other resource diffs. I'd have expected this to always come at the end?

For what it's worth, the program is:

```typescript
import * as pulumi from "@pulumi/pulumi";
import * as aws from "@pulumi/aws";
import * as path from "path";

const myBucket = new aws.s3.Bucket("my-bucket", {
    website: { /*+*/
        indexDocument: "index.html", /*+*/
    }, /*+*/
});

const myObject = new aws.s3.BucketObject("index.html", {
    acl: "public-read", /*+*/
    bucket: myBucket,
    contentType: "text/html", /*+*/
    source: path.join("site", "index.html"),
});

export const bucketName = myBucket.bucket;
export const bucketEndpoint = pulumi.interpolate`http://${myBucket.websiteEndpoint}`; /*+*/
```

where the lines with `/*+*/` are added compared to the prior update (and hence are in the diff).

### PR DESCRIPTION
**Pull request by aq17**
# Description

Fixes #3400 

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/backend/apply.go [hunk 1]
```diff
@@ -236,7 +236,20 @@ func createDiff(updateKind apitype.UpdateKind, events []engine.Event, displayOpt
...
+		// display output events last
+		if e.Type == engine.ResourceOutputsEvent {
+			outputEvents = append(outputEvents, e)
+			continue
+		}
+		msg := display.RenderDiffEvent(e, seen, displayOpts)
+		if msg != "" && e.Type != engine.SummaryEvent {
+			_, err := buff.WriteString(msg)
+			contract.IgnoreError(err)
+		}
```
Comment (dixler) [inline pkg/backend/apply.go:250]: It looks like this code is also duplicated below, but won't be run as the only `outputEvents` are `engine.ResourceOutputEvents`.

is it possible to remove the bottom code?

```go
		if msg != "" && e.Type != engine.SummaryEvent {
			_, err := buff.WriteString(msg)
			contract.IgnoreError(err)
		}
```

Comment (aq17) [reply pkg/backend/apply.go:250]: Good catch 😅

**Review by dixler — COMMENTED** [review]
code looks good, but you should add some tests to make sure that we don't break this ordering in the future.

I think there is a random pet example provider somewhere that could be helpful.

**Comment by mikhailshilkov** [comment]
Excited to see this land soon! Will let others review the actual code change.

**Comment by mikhailshilkov** [comment]
Excited to see this land soon! Will let others review the actual code change.

**Review by AaronFriel — APPROVED** [review]
LGTM

**Pull request merged**